### PR TITLE
Cleanup Source Tree

### PR DESCRIPTION
**Pull request by jasonLaster**
### Summary of Changes

* remove immutable usage
* switch to a cleaner getSource helper
* use isDirectory in as many places as possible
* cleanup render
  * move handlers out
  * move small components (project root) out

### Test Plan

* use the jest tests to guide the work
* use the mochitests

### CODE REVIEW

**Comment by darkwing** [comment]
Really excited this looks so much better.  Yesssssss.

A few thing we'll loo to fix:

1.  Tests, obviously
2.  When you set a domain as root, the contents are off by one:

offbyonedomain

Every other case loos fine though!

File: src/components/PrimaryPanes/SourcesTree.js [hunk 1]
```diff
@@ -263,6 +264,27 @@ class SourcesTree extends Component<Props, State> {
     showMenu(event, menuOptions);
   };
 
+  onExpand = (item, expandedState) => {
```
Comment (wldcordeiro) [inline src/components/PrimaryPanes/SourcesTree.js]: General comment on this file's changes, love the clean-up. It's clearer what you're doing and things are organized more logically.

**Comment by darkwing** [comment]
Current issues to resolve:

- [ ] Flow
- [ ] Setting a domain root as the project root results in double appearance of its folder:

doublefolders

**Comment by jasonLaster** [comment]
travis is green sans-mutations

**Comment by darkwing** [comment]
@jasonLaster Check out the ... gross ... update I needed to make to get this green.  I'm not seeing the source of this issue though -- do you have any insight?

**Pull request merged**